pilot/pkg/model/service.go

```diff
@@ -1228,12 +1235,28 @@ func serviceResourceName(s *workloadapi.Service) string {
 	return s.Namespace + "/" + s.Hostname
 }
 
+type ServiceScope string
+
+const (
+	// Local ServiceScope specifies that istiod will not automatically expose the matching services' endpoints at the
+	// cluster's east/west gateway. Istio will also not automatically share locolly matching endpoints with the
+	// cluster's local dataplane that are not within the local cluster.
+	Local ServiceScope = "local"
+	// Global ServiceScope specifies that istiod will automatically expose the matching services' endpoints at the
+	// cluster's east/west gateway. Istio will also automatically share globally matching endpoints with the cluster's
+	// local dataplance that in the local and remote clusters.
+	Global ServiceScope = "global"
+)
+
 type WorkloadInfo struct {
 	Workload *workloadapi.Workload
 	// Labels for the workload. Note these are only used internally, not sent over XDS
 	Labels map[string]string
 	// Source is the type that introduced this workload.
 	Source kind.Kind
+	// ScopeForService service key to ServiceScope. This mapping is required since a workload can be a part of
+	// multiple service. A single scope from a service is not sufficiently representive
```

Comment:
Not sure the last sentence is strictly necessary
```suggestion
	// multiple services.
```